index.ts

```diff
@@ -162,6 +168,33 @@ export class HttpClient {
         return this.request(verb, requestUrl, stream, additionalHeaders);
     }
 
+    /**
+     * Gets a typed object from an endpoint
+     * Be aware that not found returns a null.  Other errors (4xx, 5xx) reject the promise
+     */
+    public async getJson<T>(requestUrl: string, additionalHeaders?: ifm.IHeaders): Promise<ITypedResponse<T>> {
+        let res: ifm.IHttpClientResponse = await this.get(requestUrl, additionalHeaders);
+        return this._processResponse<T>(res, this.requestOptions);
+    }
+    
+    public async postJson<T>(requestUrl: string, obj:T, additionalHeaders?: ifm.IHeaders): Promise<ITypedResponse<T>> {
```

Comment:
This doesnt seem to work at least for actions/cache types like `ReserveCache[Request / Response]` since in input type `T = ReserveCacheRequest` DNE the return type `T = ReserveCacheResponse` 

Need to think on it more, or maybe someone else has ideas here, but it seems to me that the input type does not need to be `T` and the type parameter should only refer to the return type that we parse into. You dont gain much by specifying the input type since... its just the type you pass in. The typed-rest-client takes `any` here instead of `T`

Reply:
Changing the input type `T` =>` any` shouldnt be breaking because any `T` is assignable to `any`, if we go that route